### PR TITLE
Add Resource Detectors to payment service

### PR DESCRIPTION
**Pull request by svrnm**
# Changes

Like #648 this will add resource detectors to the payment service, to create richer data for the backend, e.g.:

Screenshot 2022-12-21 at 12 45 52

## Merge Requirements

* [X] `CHANGELOG.md` updated to document new feature additions
* [X] Appropriate documentation updates in the [docs](../docs/) folder

### CODE REVIEW

**Review by julianocosta89 — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -156,3 +156,4 @@ significant modifications will be credited to OpenTelemetry Authors.
 ([#617](https://github.com/open-telemetry/opentelemetry-demo/pull/617))
 * Use `frontend-web` as service name for browser/web requests
 ([#628](https://github.com/open-telemetry/opentelemetry-demo/pull/628))
+* Add resource detectors to payment service
```
Comment (fatsheep9146) [inline CHANGELOG.md:159]: the changelog seems not complete

Comment (svrnm) [reply CHANGELOG.md:159]: yes! Thanks

Comment (svrnm) [reply CHANGELOG.md:159]: fixed

File: src/paymentservice/package.json [hunk 2]
```diff
@@ -22,6 +22,10 @@
     "@opentelemetry/exporter-trace-otlp-grpc": "0.33.0",
     "@opentelemetry/exporter-metrics-otlp-grpc": "0.33.0",
     "@opentelemetry/sdk-node": "0.33.0",
+    "@opentelemetry/resource-detector-alibaba-cloud": "^0.27.3",
+    "@opentelemetry/resource-detector-aws": "^1.2.1",
+    "@opentelemetry/resource-detector-container": "^0.2.1",
+    "@opentelemetry/resource-detector-gcp": "^0.27.4",
```
Comment (mic-max) [inline src/paymentservice/package.json:28]: Let's be consistent with pinning to exact version numbers

```suggestion
    "@opentelemetry/resource-detector-alibaba-cloud": "0.27.3",
    "@opentelemetry/resource-detector-aws": "1.2.1",
    "@opentelemetry/resource-detector-container": "0.2.1",
    "@opentelemetry/resource-detector-gcp": "0.27.4",
```

**Review by mic-max — APPROVED** [review]

**Comment by fatsheep9146** [comment]
please fix conflicts, lgtm :)

**Review by fatsheep9146 — APPROVED** [review]

**Review by puckpuck — APPROVED** [review]

**Pull request merged**